src/plugins/workspace/public/components/workspace_creator/workspace_form_summary_panel.tsx

```diff
@@ -0,0 +1,201 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import React, { useCallback, useState } from 'react';
+import {
+  EuiCard,
+  EuiFlexGroup,
+  EuiFlexItem,
+  EuiIcon,
+  EuiSpacer,
+  EuiText,
+  EuiTextColor,
+  EuiLink,
+} from '@elastic/eui';
+import { i18n } from '@osd/i18n';
+import { WorkspaceFormDataState } from '../forms';
+import { WorkspaceUseCase } from '../../types';
+import { RightSidebarScrollField, RIGHT_SIDEBAR_SCROLL_KEY } from './utils';
+
+export const FieldSummaryItem = ({
+  field,
+  children,
+  bottomGap = true,
+}: React.PropsWithChildren<{
+  field: RightSidebarScrollField;
+  bottomGap?: boolean;
+}>) => {
+  const handleTitleClick = useCallback(() => {
+    const element = document.querySelector(
+      `.workspaceCreateFormContainer [${RIGHT_SIDEBAR_SCROLL_KEY}="${field}"]`
+    );
+
+    element?.scrollIntoView({ behavior: 'smooth', block: 'center' });
+  }, [field]);
+
+  const renderTitle = () => {
+    return {
+      [RightSidebarScrollField.UseCase]: i18n.translate(
+        'workspace.form.summary.panel.useCase.title',
+        {
+          defaultMessage: 'Use case',
+        }
+      ),
+      [RightSidebarScrollField.Name]: i18n.translate('workspace.form.summary.panel.name.title', {
+        defaultMessage: 'Name',
+      }),
+      [RightSidebarScrollField.Description]: i18n.translate(
+        'workspace.form.summary.panel.description.title',
+        {
+          defaultMessage: 'Description',
+        }
+      ),
+      [RightSidebarScrollField.Color]: i18n.translate('workspace.form.summary.panel.color.title', {
+        defaultMessage: 'Accent color',
+      }),
+      [RightSidebarScrollField.DataSource]: i18n.translate(
+        'workspace.form.summary.panel.dataSources.title',
+        {
+          defaultMessage: 'Data sources',
+        }
+      ),
+      [RightSidebarScrollField.Member]: i18n.translate(
+        'workspace.form.summary.panel.members.title',
+        {
+          defaultMessage: 'Members',
+        }
+      ),
+    }[field];
```

Comment:
Nit: maybe can just make it a constant, then use it at where it's needed with `SCROLL_FIELDS[field]`
```suggestion
    const SCROLL_FIELDS =  {
      [RightSidebarScrollField.UseCase]: i18n.translate(
        'workspace.form.summary.panel.useCase.title',
        {
          defaultMessage: 'Use case',
        }
      ),
      [RightSidebarScrollField.Name]: i18n.translate('workspace.form.summary.panel.name.title', {
        defaultMessage: 'Name',
      }),
      [RightSidebarScrollField.Description]: i18n.translate(
        'workspace.form.summary.panel.description.title',
        {
          defaultMessage: 'Description',
        }
      ),
      [RightSidebarScrollField.Color]: i18n.translate('workspace.form.summary.panel.color.title', {
        defaultMessage: 'Accent color',
      }),
      [RightSidebarScrollField.DataSource]: i18n.translate(
        'workspace.form.summary.panel.dataSources.title',
        {
          defaultMessage: 'Data sources',
        }
      ),
      [RightSidebarScrollField.Member]: i18n.translate(
        'workspace.form.summary.panel.members.title',
        {
          defaultMessage: 'Members',
        }
      ),
    };
```

Reply:
Looks good to me, will update.